### PR TITLE
Update API-Designer to expose CSDL-XML tab.

### LINKED ISSUE #357: Enable generating XML CSDL from RSDL in Javascript
Our .NET rsdl library supports generating either XML CSDL or JSON CSDL from RSDL. However, our javascript library appears only to support generating JSON CSDL.

Existing workflows that use XML CSDL may want to leverage RSDL and RSDL tooling.  To do so, we either need:
1) The ability to generate XML CSDL from RSDL, or
2) The ability to generate XML CSDL from JSON CSDL.

### LINKED ISSUE #402: add csdl XML tab to RAPID API designer
image

### PR DESCRIPTION
**Pull request by mikepizzo**
Fixes #357 
Fixes #402

Adds a simple JSON CSDL->XML CSDL serializer (in csdl-xml subdirectory of rsdl).
todo:
- [x] Add unit tests
- [ ] Write non-string annotations
- [ ] Write referential constraints
- [x] Write TypeDefinitions
- [ ] Improve error handling
- [ ] Update to xmlbuilder2
- [x] Consider moving csdl-xml out of rsdl directory
- [ ] Formatting of XML pane in api-designer

### CODE REVIEW

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 1]
```diff
@@ -0,0 +1,653 @@
+var builder = require('xmlbuilder');
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:1]:
```suggestion
const builder = require('xmlbuilder');
```

Please use `const` for immutable variables, and `let` for ordinary variables.

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+  // <edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+  //   <edmx:DataServices>
+
+  var xmlElement = builder.begin().dec()
+    .ele('edmx:Edmx')
+      .att('xmlns:edmx','http://docs.oasis-open.org/odata/ns/edmx')
+      .att('Version','4.0');
+
+  // Write References
+  if(jsonCsdl.$References != undefined)
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:18]:
```suggestion
  if(jsonCsdl.$References)
```

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+          writeEnumType(schema, key, modelElement);
+          break;
+        case 'TypeDefinition':
+          writeTypeDefinition(schema, key, modelElement);
+          break;
+        case 'EntityContainer':
+          writeEntityContainer(schema, key, modelElement);
+          break;
+        default:
+          console.log("Unexpected Schema Element" + key);
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:67]:
```suggestion
          console.log(`Unexpected Schema Element ${key}`);
```
Use string templates?

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+  //     <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="Jetsons">
+  var dataServiceElement = xmlElement.ele('Schema');
+  var schema = dataServiceElement
+    .att('xmlns','http://docs.oasis-open.org/odata/ns/edm')
+    .att('Namespace', namespace);
+
+    var model = jsonCsdl[namespace];
+  // write model elements
+  Object.keys(model).forEach((key)=>
+  {
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:33]:
```suggestion
  for(const key of Object.keys(model)) {
```

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 5]
```diff
@@ -0,0 +1,653 @@
...
+          break;
+        case 'EntityContainer':
+          writeEntityContainer(schema, key, modelElement);
+          break;
+        default:
+          console.log("Unexpected Schema Element" + key);
+      }
+    }
+  }
+  );
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:71]:
```suggestion
```
If you switch to `for(...of...)`

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 6]
```diff
@@ -0,0 +1,653 @@
...
+function writeEntityType(parentElement, name, type)
+{
+  // write Entity element
+  // <EntityType Name="Employee">
+  var entity = parentElement.ele('EntityType')
+    .att('Name', name);
+
+  // write Key element first
+  var keys = type.$Key;
+  if (keys != undefined && keys.length > 0)
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:88]:
```suggestion
  if (keys?.length > 0)
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 7]
```diff
@@ -629,7 +628,7 @@ function writeAnnotation(xmlElement, term, annotationValue)
       break;
     break;
       default:
-        console.log("Annotation type " + typeof(annotationValue) + " not supported for term " + term);
+        console.log("Annotation type ${typeof(annotationValue)} not supported for term ${term}.");
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:631]:
```suggestion
        console.log(`Annotation type ${typeof(annotationValue)} not supported for term ${term}.`);
```
Need to use backticks for string templates

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 8]
```diff
@@ -64,11 +65,10 @@ function serializeToXml(jsonCsdl, pretty = true)
           writeEntityContainer(schema, key, modelElement);
           break;
         default:
-          console.log("Unexpected Schema Element" + key);
+          console.log("Unexpected Schema Element ${key}");
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:68]:
```suggestion
          console.log(`Unexpected Schema Element ${key}`);
```

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 9]
```diff
@@ -349,30 +349,30 @@ function writeEntitySetOrSingleton(modelElement, entitySetOrSingleton)
           }
           else
           {
-            console.log("Unsupported property on entity set or singleton: " + key);
+            console.log("Unsupported property on entity set or singleton ${key}.");
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:352]:
```suggestion
            console.log(`Unsupported property on entity set or singleton ${key}.`);
```

File: tools/rsdl/csdl-xml/lib/xmlSerializer.js [hunk 10]
```diff
@@ -541,17 +541,17 @@ function writeOperation(operationElement, operation)
         }
         else
         {
-          console.log("Not Implemented Operation Attribute: " + key);
+          console.log("Not Implemented Operation Attribute ${key}.");
```
Comment (ralfhandl) [inline tools/rsdl/csdl-xml/lib/xmlSerializer.js:544]:
```suggestion
          console.log(`Not Implemented Operation Attribute ${key}.`);
```

**Comment by ralfhandl** [comment]
Waiting for the unit tests 😄

File: tools/csdl-xml/test/cli.test.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+describe("CLI", () => {
+  it("Help", async () => {
+    const result = await cmd(["-h"]);
+    expect(result.code).to.equal(0);
+    expect(result.stdout).to.contain("Usage: csdl2xml");
+  });
+
+  it("Translate one file", async () => {
+    const outfile = "resources/jetsons.csdl.xml";
+    const basefile = "resources/jetsons.csdl.xml.base";
```
Comment (ralfhandl) [inline tools/csdl-xml/test/cli.test.js:16]: Why `.xml.base` and not `.base.xml`? That would make it easier for an XML editor to recognize the file. And with VS Code extension `redhat.vscode-xml` installed and an XML catalog configured one even gets XML schema check for the "base" files for free.

File: tools/rsdl/rsdl-js/package-lock.json [hunk 12]
```diff
@@ -1,3861 +0,0 @@
-{
```
Comment (ralfhandl) [inline tools/rsdl/rsdl-js/package-lock.json:1]: Do you really want to delete this? It belongs to a different package.

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: tools/rsdl/rsdl-js/package-lock.json (hunk 12 above)
Comment (mikepizzo) [reply tools/rsdl/rsdl-js/package-lock.json:1]: Didn't mean to remove permanently; I was getting a bunch of errors trying to build that went away when I deleted this file.  I can try and restore and fix the issues, but removing the file was the most expedient way to get back to the issue at hand.

Comment (ralfhandl) [reply tools/rsdl/rsdl-js/package-lock.json]: In such situations just delete it and run `npm install` again to restore a better version of it.

**Pull request merged**